frontend/gist-frontend/src/api/users-handler.ts

```diff
@@ -1,25 +1,39 @@
-import { User, Username } from "./api-types";
+import { User } from "./api-types";
 import { APIHandler } from "./api-handler";
 
 export class UsersHandler {
   async getUser(username: string): Promise<User> {
-    let data = await APIHandler(
-      `users/getUser`,
-      "POST",
-      new Username(username)
-    );
-    let user = new User(data);
-    return user;
+    let data = await APIHandler(`users/getUser/${username}`, "GET");
+    return new User(data);
   }
   async getAllUsers(): Promise<User[]> {
     let users: User[] = [];
     let data = await APIHandler(`users/`, "GET");
-    if (data !== undefined) {
+    if (!!data) {
       let dataArray = JSON.parse(JSON.stringify(data));
       for (let i = 0; i < dataArray.length; i++) {
         users.push(new User(dataArray[i]));
       }
     }
     return users;
   }
+  async createUser(user: User): Promise<User> {
+    let data = await APIHandler(`users/`, "POST", user);
+    return new User(data);
+  }
+  async login(username: string, password: string) {
+    let data = await APIHandler(`users/login`, "POST", {
+      username: username,
+      password: password
+    });
+    return data;
+  }
```

Comment:
So what's different about this versus the getUser call? I'm assuming this is called when a user logs in to the app and thus their user profile is fetched, so wouldn't that just be a call to getUser for all their information?

Reply:
yes, this will be called on login. the backend login function compares the password the user inputs to the hashed password stored in the db, and only returns the user if the hashes match. the getUser just returns a user for a given username without checking passwords/authenticating the user.

Reply:
aaaahhhhhhh...gotcha